docs/modules/deployment/pages/upgrade/basic.adoc

```diff
@@ -0,0 +1,134 @@
+PostgreSQL's `pg_upgrade` utility copies the existing database into the new database and does not alter the existing database's data files.
+This leaves the PostgreSQL 9.4 database in its original state and location which can be used for rollback.
+
+. Replace `/var/lib/pgsql/12/data/postgresql.conf` with the updated copy.
+. Copy `pg_hba.conf` from the postgresql 9.4 data directory to the postgresql-12 data directory: `cp /var/lib/pgsql/data/pg_hba.conf /var/lib/pgsql/12/data/pg_hba.conf`
++
+NOTE: Verify these paths align with reality.
+. Exit PostgreSQL shell: `exit`
++
+Expected outcome: `whoami` returns `root`
+
+== Update and verify the {page-component-title} repository
+
+Your {page-component-title} repository is defined in a file in `/etc/yum.repos.d/`.
+It may be named {page-component-title}.repo  but is not guaranteed to be.
+
+. Use the text editor of your choice to edit your {page-component-title} repository config to change the `baseurl` of the repository from the old release family to the new release family.
+. Verify that the attribute `enabled=1` is defined, or enabled, or absent entirely
+. Purge any cached yum data: `yum clean all`
++
+Expected outcome: `yum repolist` does not show an error pertaining to the {page-component-title} repo; `yum info {page-component-title}-core` shows an available current package version.
+
+. Install the PostgreSQL 12 version of iplike: `yum -y install iplike-pgsql12`
++
+Expected outcome: `rpm -q iplike-pgsql12` reports an _iplike-pgsql12_ package is installed.
+
+. Start postgresql-12: `systemctl start postgresql-12.service`
++
+Expected outcome: `systemctl status postgresql-12.service` reports postgresql-12 has started.
+
+. Make a backup copy of your OpenNMS config: `mv /opt/opennms/etc /opt/opennms/etc.backup`
+
+. Upgrade the {page-component-title} packages to the new version: `yum -y upgrade {page-component-title}-core {page-component-title}-webapp-jetty`
++
+Expected result: A lot of rpmnew files; `rpm -q {page-component-title}-core` reports the new version is installed.
+
+. Move the package-updated etc directory out of the way: `mv $OPENNMS_HOME/etc $OPENNMS_HOME/etc.as.shipped.2021`
+
+. ##What do users without support tarball do?## Unpack the OpenNMS Support-provided config tarball in `/opt/opennms`: `cd /opt/opennms; tar zxvf apple_etc.tar.gz`
++
+Expected result: a new /etc directory is created.
+
+. Install Java 11: `yum -y install java-11-openjdk java-11-openjdk-devel`
++
+Expected result: `rpm -q java-11-openjdk` returns a result.
+
+. Execute runjava:
+`$OPENNMS_HOME/bin/runjava -S /usr/lib/jvm/java-11/bin/java`
++
+Expected outcome: `$OPENNMS_HOME/etc/java.conf` contains a path to the Java 11 JDK.
+
+. Run the opennms installer: `$OPENNMS_HOME/bin/install -dis`
++
+NOTE: This step took approximately two hours in a lab environment processing 1.9 million database events.
+It may be faster on your own hardware, but may still take significant time.
```

Comment:
```suggestion
```

This only comes into play upgrading from something 20-or-older to 21-or-newer (iirc), because there is a migration of event data into two separate database tables, s it has to cycle through every event.  Most upgrades will not encounter this.